sctp: update manpages

llfw opened the pull request:
Add missing documentation for sctp_sendv(3) and sctp_recvv(3).

Add a note that sctp_send(3), sctp_sendx(3) and sctp_recvmsg(3) are deprecated by RFC 6458.

Add a STANDARDS section to all functions specified in RFC 6458 to indicate their standards conformance.

---

this should probably be reviewed by someone who knows more about SCTP than i do.

igalic commented:
cc @kprovost @tuexen

tuexen commented on lib/libc/net/sctp_send.3:
```diff
@@ -48,11 +49,25 @@
 is used to transmit a message to another SCTP endpoint.
+The
+.Fn sctp_sendx
+function is used to transmit a message to multiple endpoint addresses.
```
Maybe something like
When issuing an implicit association setup, the
.Fn sctp_sendx
function allows to specify multiple addresses of the peer, which can then be used for the association setup.
This is equivalent to using
.Xr sctp_connectx 3 .

llfw replied:
i replaced the text with this, which i think is still accurate, but is slightly shorter to fit the style of the paragraph:

```
The
.Fn sctp_sendx
function is used to specify multiple peer addresses when creating an implicit
association, as in
.Xr sctp_connectx 3 .
```

tuexen commented:
Looking at, for example [IP6(4)](https://man.freebsd.org/cgi/man.cgi?query=ip6&apropos=0&sektion=0&manpath=FreeBSD+14.0-RELEASE+and+Ports&arch=default&format=html#end), it might be useful to add references to RFC 6458 in the `SEE ALSO` section to the man pages  where you added a `STANDARDS` section.

bsdbcr commented on lib/libc/net/sctp_recvmsg.3:
```diff
@@ -42,22 +43,28 @@
-successful
+and
+.Fn sctp_recvv
+functions are used to receive a message from another SCTP endpoint.  They are
```
You need to have a line break after the sentence stop here.

bsdbcr commented on lib/libc/net/sctp_recvmsg.3:
```diff
@@ -231,6 +238,158 @@ The
+.Fn sctp_recvv
+function works as
+.Fn sctp_recvmsg
+with two differences.  Firstly, the receive buffer is passed as an array
```
Another line break is needed here when the first sentence ends.

bsdbcr commented:
The man page changes need a line break after a sentence stop. Running "mandoc -Tlint" and the textproc/igor port find these sort of issues.

llfw commented:
- fixed lint errors
- added .Rs for RFC 6458
- changed the description of sctp_sendx
- rebased on current main (although there were no conflicts here)

bsdbcr commented:
The man page changes look good now, thank you! 👍
